cores/esp32/Stream.h

```diff
@@ -40,8 +40,8 @@ class Stream: public Print
 protected:
     unsigned long _timeout;      // number of milliseconds to wait for the next char before aborting timed read
     unsigned long _startMillis;  // used for timeout measurement
-    int timedRead();    // private method to read stream with timeout
-    int timedPeek();    // private method to peek stream with timeout
+    virtual int timedRead();    // private method to read stream with timeout
+    virtual int timedPeek();    // private method to peek stream with timeout
```

Comment:
this may break compatibility with libraries.

Reply:
@atanisoft can you be more precise and tell how will it break things? any usage when timedRead should be called directly on Stream.h without virtual inheritance?

Reply:
Stream.h is defined by the Arduino core specification, you can see the declaration of these methods [here](https://github.com/arduino/ArduinoCore-avr/blob/master/cores/arduino/Stream.h#L54-L55). By adding virtual to these methods it will potentially break compatibility with any library which overrides these methods without the declaration of virtual. A similar problem has been seen in https://github.com/espressif/arduino-esp32/issues/2755

Reply:
I cannot imagine any viable scenario about how virtual can break the things. I've got your points about UART HAL methods, and will workaround them. But it is useless if you discard this PR because of virtual methods. Let's decide if this have chance to be merged?

@me-no-dev, what's your opinion about this ?

Reply:
I'm not against the addition of virtual to the declaration but it may have unintended breakages in libraries which extend this class